<launch>
   
   <arg name="port_name" default="/dev/ttyUSB0"/>
   <arg name="port_rate" value="115200" />
   <arg name="sensor_bias" value="10" />
   <arg name="ns" value="/" />
   <arg name="sensor_threshold" value="20" />



   <node pkg="roshand_gen2_driver" type="roshand_gen2_bringup" name="roshand_gen2_bringup" output="screen" required="true" cwd="node" >
      <param name="port_name" value="$(arg port_name)" />  
      <param name="port_rate" value="$(arg port_rate)" />
      <param name="ns" value="$(arg ns)" />
      <param name="sensor_bias" value="$(arg sensor_bias)" />
      <param name="sensor_threshold" value="$(arg sensor_threshold)" />

   </node>     

   <include file="$(find roshand_gen2_visualizer)/launch/display.launch" />

   <node name = "roshand_state_publisher"   pkg="robot_state_publisher"    type="robot_state_publisher">
      <remap from="joint_states" to="$(arg ns)RosHand_JointState"/>
   </node>

  <node name="sensor_marker" pkg="roshand_gen2_driver" type="sensor_marker" >
      <param name="ns" value="$(arg ns)" />

  </node>
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find roshand_gen2_visualizer)/roshand_gen2.rviz" />

</launch>
